xsd-core/src/main/java/io/ballerina/xsd/core/XSDToRecord.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.xsd.core;
+
+import io.ballerina.compiler.syntax.tree.ModuleMemberDeclarationNode;
+import io.ballerina.compiler.syntax.tree.ModulePartNode;
+import io.ballerina.compiler.syntax.tree.NodeParser;
+import io.ballerina.xsd.core.visitor.XSDVisitor;
+import io.ballerina.xsd.core.visitor.XSDVisitorImpl;
+import io.ballerina.xsd.core.component.XSDComponent;
+import io.ballerina.xsd.core.visitor.VisitorUtils;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import static io.ballerina.xsd.core.visitor.VisitorUtils.CLOSE_BRACES;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.COMMA;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.OPEN_BRACES;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.QUOTATION_MARK;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.WHITESPACE;
+import static io.ballerina.xsd.core.visitor.XSDVisitorImpl.CONTENT_FIELD;
+import static io.ballerina.xsd.core.visitor.XSDVisitorImpl.ENUM;
+import static io.ballerina.xsd.core.visitor.XSDVisitorImpl.NAME;
+import static io.ballerina.xsd.core.visitor.XSDVisitorImpl.RECORD_WITH_OPEN_BRACE;
+import static io.ballerina.xsd.core.visitor.XSDVisitorImpl.SEMICOLON;
+import static io.ballerina.xsd.core.visitor.XSDVisitorImpl.VERTICAL_BAR;
+
+/**
+ * This class is used for transforming an XSD into a corresponding record format.
+ *
+ * @since 0.1.0
+ */
+public final class XSDToRecord {
+    public static final String SCHEMA = "schema";
+    public static final String EOF_TOKEN = "";
+    public static final String INVALID_IMPORTS_ERROR = "Invalid imports have been found.";
```

Comment:
Dont we need to move this into a `DiagnosticErrorLog` class, like we do in stdlib modules

Reply:
Fixed with 43067b6